### PR TITLE
Add the ability to publish from local project

### PR DESCRIPTION
**Pull request by SChandarana**
Using Environment variables for discord token and google sheets credentials the project can be published to azure and ReleaseBot works (using the information from Azure)

Still yet to do the secret manager stuff as I can't seem to get my head around it and how it would properly work with azure. Right now just using launchsettings.json for local dev and not commiting it (as it isn't actually needed for this project).

Will have the others test Publishing from their own machines, which should be possible by Right-Clicking the DiscordBot project and hitting publish

### CODE REVIEW

File: DiscordBot.DataAccess/EventsSheetsService.cs [hunk 1]
```diff
@@ -32,7 +31,7 @@ public class EventsSheetsService : IEventsSheetsService
     {
         private static readonly string[] scopes = { SheetsService.Scope.Spreadsheets };
         private static readonly string applicationName = "Softwire Discord Bot";
-        private static readonly string spreadsheetId = "";
+        private static readonly string spreadsheetId = Environment.GetEnvironmentVariable("GOOGLE_SHEET_ID");
```
Comment (FridaTveit) [inline DiscordBot.DataAccess/EventsSheetsService.cs:34]: I'd recommend storing the names of these variables somewhere, e.g. a static class with several const strings 🙂

File: DiscordBot.DataAccess/EventsSheetsService.cs [hunk 2]
```diff
@@ -208,15 +207,21 @@ public async Task<IEnumerable<DiscordEvent>> ListEventsAsync()
...
-            return GoogleCredential.FromStream(stream)
-                       .CreateScoped(scopes)
-                       .UnderlyingCredential as ServiceAccountCredential
-                   ?? throw new EventsSheetsInitialisationException("Credential maker returned null");
+            var clientEmail = Environment.GetEnvironmentVariable("GOOGLE_CLIENT_EMAIL");
+            var projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+            var privateKeyId = Environment.GetEnvironmentVariable("GOOGLE_PRIVATE_KEY_ID");
+            var privateKey = Environment.GetEnvironmentVariable("GOOGLE_PRIVATE_KEY");
+            return new ServiceAccountCredential(
+                new ServiceAccountCredential.Initializer(clientEmail)
```
Comment (FridaTveit) [inline DiscordBot.DataAccess/EventsSheetsService.cs:217]: This is a bit hard to read. I'd recommend creating the initialiser first then the credential 🙂

File: DiscordBot/Settings.job [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (FridaTveit) [inline DiscordBot/Settings.job:1]: What's this used for? 🙂

**Review by FridaTveit — COMMENTED** [review]
I've left a few comments but otherwise looks good! Some documentation for how to do a manual release would definitly be useful, but that might be part of what @timleachsw is writing? 🙂

**Comment by timleachsw** [comment]
> 
> 
> I've left a few comments but otherwise looks good! Some documentation for how to do a manual release would definitly be useful, but that might be part of what @timleachsw is writing? 🙂

@FridaTveit I'll make sure to include this in the README yeah!

**Comment by benjimarshall** [comment]
@SChandarana asked me to do some of these bits because he's going to be spending the next little while debugging his laptop power issues (it appears its the port in his laptop that's the issue, and not the charger)

File: DiscordBot.DataAccess/EventsSheetsService.cs (hunk 2 above)
Comment (benjimarshall) [reply DiscordBot.DataAccess/EventsSheetsService.cs]: Done in 4d9b757 🙂

File: DiscordBot.DataAccess/EventsSheetsService.cs (hunk 1 above)
Comment (benjimarshall) [reply DiscordBot.DataAccess/EventsSheetsService.cs]: Done in 4d9b757 🙂

An alternate solution that I came up with was to extract all of the `Environment.GetEnvironmentVariable(...)` out into a static class, along the line of:

<details>
<summary>Spoiler</summary>

```C#
internal static class SheetsEnvironmentVariables
{
    const string sheetIdVar = "GOOGLE_SHEET_ID";
    const string clientEmailVar = "GOOGLE_CLIENT_EMAIL";
    const string projectIdVar = "GOOGLE_PROJECT_ID";
    const string privateKeyIdVar = "GOOGLE_PRIVATE_KEY_ID";
    const string privateKeyVar = "GOOGLE_PRIVATE_KEY";

    public static string? SheetId => Environment.GetEnvironmentVariable(sheetIdVar);
    public static string? ClientEmail => Environment.GetEnvironmentVariable(clientEmailVar);
    public static string? ProjectId => Environment.GetEnvironmentVariable(projectIdVar);
    public static string? PrivateKeyId => Environment.GetEnvironmentVariable(privateKeyIdVar);
    public static string? PrivateKey => Environment.GetEnvironmentVariable(privateKeyVar);
}
```
</details>

What do you think? 🙂

Comment (FridaTveit) [reply DiscordBot.DataAccess/EventsSheetsService.cs]: I like it 🙂

Comment (benjimarshall) [reply DiscordBot.DataAccess/EventsSheetsService.cs]: Added in 427658b 🙂

File: DiscordBot/Settings.job (hunk 3 above)
Comment (benjimarshall) [reply DiscordBot/Settings.job]: It's a redundant config file. Removed c98cdf6 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**